hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SequenceIDGenerator.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * <p>Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.hadoop.hdds.scm.ha;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * SequenceIDGenerator uses higher 30 bits to save the term, and lower 34 bits
+ * to save a count (increase from 0). Each call of nextID() will increase the
+ * count on the lower 34 bits by 1, thus SequenceIDGenerator always generates
+ * unique number within a specific term.
+ */
+public class SequenceIDGenerator {
+  private final long curTermOnHigher30Bits;
+  private final AtomicLong counter = new AtomicLong(0L);
+
+  public SequenceIDGenerator(long term) {
+    // move term to higher 30 bits and save it into curTermOnHigher30Bits.
+    curTermOnHigher30Bits = term << 34;
+  }
+
+  public long nextID() {
+    long l = counter.getAndIncrement();
+    return l & 0x3FFFFFFF | curTermOnHigher30Bits;
```

Comment:
I have thought about whether we need a check here to see if the next unique ID exceeds the 34 bits (thus become non-unique). My current thought is no need to add such a check:

34 bits can give 1,735,166,787,583 (>1.7 trillion) numbers for a term. It is very very very less likely that another leader election won't happen for the specific term. Meanwhile, adding a check means we will need to do >1.7 trillion checks before hitting the rare case.

Combine both I think we don't need to do that check here (save the cost as the rare case will not happen).

Reply:
Thanks @amaliujia for the work !

The up limit of 34 bits is 17 billion, and I consider we have to do such a check, and need to exit SCM when it reaches the limit. Since customer might deploy a single-server raft cluster in the non-HA mode, there will be no leader election after SCM starts.

Reply:
and it should be `l & 0x3FFFFFFFF`, seems missing one F

Reply:
Yes your are right that I have missed a `F`. 

Also add the check to prevent non unique ID generated.